### ISSUE TITLE
Add role and aria-label (a11y)

### ISSUE THREAD

**Issue by juuso-vaadin** [vaadin-component-factory/popup#19]
For optimal accessibility, the component should have `role="dialog"` set for the `vcf-popup-overlay` element.

[Dialogs should always have a label.](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/dialog_role) Labelling can be done with `aria-label` property on the `vcf-popup-overlay` element. Vaadin dialog has a feature which sets the aria-label when `setHeaderTitle()` method is used. In addition it should be possible to set aria-label manually in case visible header is not used.

**Comment by bwajtr** [comment]
Fixed with the following commit:
https://github.com/vaadin-component-factory/popup/commit/73e1932dfd570e56fa44a02d2788b7745c0db23e

**Comment by bwajtr** [comment]
Released in version 23.3.3 and 24.0.4
